src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -23,15 +23,17 @@ public class Memory : Indexable.Indexable, IMemory {
     /// <summary>
     /// Instantiate a new memory bus.
     /// </summary>
+    /// <param name="memoryBreakpoints">The class that manages breakpoints based on memory access.</param>
     /// <param name="baseMemory">The memory device that should provide the default memory implementation</param>
-    /// <param name="is20ThAddressLineSilenced">whether the A20 gate is silenced.</param>
-    public Memory(IMemoryDevice baseMemory, bool is20ThAddressLineSilenced) {
+    /// <param name="a20gate">The class that implements A20 Gate on/off support.</param>
+    public Memory(MemoryBreakpoints memoryBreakpoints, IMemoryDevice baseMemory, A20Gate a20gate) {
```

Comment:
How about a an empty collection ? (as an optional parameter)

Reply:
I'd rather not introduce nullable case(s) handling unless there's a clear cut advantage.

I feel that since this is a reverse engineering emulator, creating Memory should not be possible without memory breakpoints.